components/raftstore/src/store/worker/compact.rs

```diff
@@ -127,6 +130,30 @@ where
         Runner { engine }
     }
 
+    /// Periodic full compaction.
+    ///
+    /// NOTE this is a highly experimental feature!
+    ///
+    /// TODO: Do not start if there is heavy I/O.
+    /// TODO: Make it possible to rate limit, pause, or abort this by compacting
+    /// a range at a time.
+    pub fn full_compact(&mut self) -> Result<(), Error> {
+        fail_point!("on_full_compact");
+        let timer = Instant::now();
+        let full_compact_timer = FULL_COMPACT.start_coarse_timer();
+        box_try!(self.engine.compact_range(
+            None, None, // Compact the entire key range.
+            true, // exclusive manual: do not run if background compaction is running
+            1,    // number of threads threads
+        ));
+        full_compact_timer.observe_duration();
+        info!(
+            "full compaction finished";
+            "time_takes" => ?timer.saturating_elapsed(),
+        );
```

Comment:
there's no harm to add the log when it get started.

Reply:
done.